src/server/node.rs

```diff
@@ -114,6 +113,7 @@ where
     pub fn start<T>(
         &mut self,
         engines: Engines,
+        engine_snapshot: Arc<RwLock<Option<SyncSnapshot>>>,
```

Comment:
can we move this to Engines, seem it is weird to pass an Engines and an engine snapshot together.